compiler/noirc_evaluator/src/brillig/brillig_ir/codegen_stack.rs

```diff
@@ -237,117 +232,151 @@ mod tests {
+        assert_eq!(opcodes.len(), 8);
+    }
+
+    #[test]
+    fn test_another_loop_with_branch() {
+        let movements = vec![(2, 1), (1, 2), (2, 3), (3, 4), (4, 5)];
+        let mut context = create_context();
+
+        let (sources, destinations) = movements_to_source_and_destinations(movements);
 
         context.codegen_mov_registers_to_registers(sources, destinations);
         let opcodes = context.artifact().byte_code;
         assert_eq!(
             opcodes,
             vec![
                 Opcode::Mov {
-                    destination: MemoryAddress::relative(1),
-                    source: MemoryAddress::relative(10)
+                    destination: MemoryAddress::relative(5),
+                    source: MemoryAddress::relative(4)
                 },
                 Opcode::Mov {
-                    destination: MemoryAddress::relative(10),
-                    source: MemoryAddress::relative(13)
+                    destination: MemoryAddress::relative(4),
+                    source: MemoryAddress::relative(3)
                 },
                 Opcode::Mov {
-                    destination: MemoryAddress::relative(13),
-                    source: MemoryAddress::relative(12)
+                    destination: MemoryAddress::relative(3),
+                    source: MemoryAddress::relative(2)
                 },
                 Opcode::Mov {
-                    destination: MemoryAddress::relative(12),
-                    source: MemoryAddress::relative(11)
+                    destination: MemoryAddress::relative(2),
+                    source: MemoryAddress::relative(1)
                 },
                 Opcode::Mov {
-                    destination: MemoryAddress::relative(11),
-                    source: MemoryAddress::relative(1)
+                    destination: MemoryAddress::relative(1),
+                    source: MemoryAddress::relative(3)
                 }
             ]
         );
     }
-
     #[test]
-    fn test_mov_registers_to_registers_loop_and_branch() {
-        let movements = vec![(10, 11), (11, 12), (12, 10), (10, 13), (13, 14)];
-        let (sources, destinations) = movements_to_source_and_destinations(movements);
+    fn test_one_loop() {
+        let movements = vec![(2, 1), (4, 2), (5, 3), (3, 4), (1, 5)];
         let mut context = create_context();
+        for _ in 0..movements.len() {
+            context.registers_mut().allocate_register();
+        }
+        let (sources, destinations) = movements_to_source_and_destinations(movements);
 
         context.codegen_mov_registers_to_registers(sources, destinations);
         let opcodes = context.artifact().byte_code;
         assert_eq!(
             opcodes,
             vec![
                 Opcode::Mov {
-                    destination: MemoryAddress::relative(1),
-                    source: MemoryAddress::relative(10)
-                }, // Temporary
+                    destination: MemoryAddress::relative(6),
```

Comment:
Could we bring back the comment about how this is a move to a temporary register

Reply:
Done